src/main/java/com/amazon/opendistroforelasticsearch/sql/rewriter/nestedfield/NestedFieldProjection.java

```diff
@@ -123,23 +135,64 @@ private void buildInnerHit(List<String> fieldNames, NestedQueryBuilder query) {
      * Why linear search? Because NestedQueryBuilder hides "path" field from any access.
      * Assumption: collected NestedQueryBuilder list should be very small or mostly only one.
      */
-    private NestedQueryBuilder findNestedQueryWithSamePath(List<NestedQueryBuilder> nestedQueries, String path) {
+    private NestedQueryBuilder findNestedQueryWithSamePath(
+            List<NestedQueryBuilder> nestedQueries, String path, List<String> fieldNames) {
+
         return nestedQueries.stream().
-                             filter(query -> isSamePath(path, query)).
-                             findAny().
-                             orElseGet(createEmptyNestedQuery(path));
+                filter(query -> isSamePath(path, query, fieldNames)).
+                findAny().
+                orElseGet(createNestedQuery(path, fieldNames));
+    }
+
+    /** Check against NestedQuery of WHERE condition */
```

Comment:
Please add better explanation, what exactly is checked

Reply:
After removing the inner `if` condition. I'll also remove the documentation as well.